src/lib.rs

```diff
@@ -44,8 +44,33 @@ impl<'a> Iterator for ErrorIterator<'a> {
 
         None
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let mut count = 0_usize;
+        let mut next = self.inner;
+        while let Some(error) = next {
```

Comment:
Since `size_hint` is now iterating the remainder of the chain, I don't know if that could be called an optimization (using terminology from the `size_hint` documentation).

What you have here is the correct thing to do for `ExactSizeIterator`, however.